intermediate_notebooks/E2E/mortgage/mortgage_e2e.ipynb

```diff
@@ -86,30 +88,101 @@
     "IPADDR = str(output.decode()).split()[0]\n",
     "\n",
     "cluster = LocalCUDACluster(ip=IPADDR)\n",
+    "#client = Client(address=\"10.136.7.103:8786\")\n",
     "client = Client(cluster)\n",
     "client"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Use `pynvml` to determine the GPU memory and decide `start_year`, `end_year` and `part_count`"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# FUNCTION ASSUMES ALL GPUS ARE IDENTICAL WITHIN A NODE\n",
+    "def memory_info():\n",
+    "    pynvml.nvmlInit()\n",
+    "    handle = pynvml.nvmlDeviceGetHandleByIndex(0)\n",
+    "    gpu_mem = pynvml.nvmlDeviceGetMemoryInfo(handle).total\n",
+    "    pynvml.nvmlShutdown()\n",
+    "    return (gpu_mem)\n",
+    "\n",
+    "res = client.run(memory_info)\n",
+    "total_mem = sum(res.values())\n",
+    "min_mem = min(res.values())"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# ETL memory usage depends on GPU memory per card (not number of GPU's). \n",
+    "# Assumption is 1gb split variant of the data is used\n",
+    "\n",
+    "start_year = None # start year for etl proessing\n",
```

Comment:
Text about logic to automatically select year and part_count based on gpu mem and to change as required.
(Current values for years and part_count seems a bit conservative and can be updated)

Reply:
Is this still an open question?

Reply:
Yes. Let me clarify:
- Should we have a md cell explaining what the aim is and to encourage the user to change these values based on their requirements.
- The current values are conservative and should work. Do we instead modify it to the upper_limit of data it could process without OOM for the cases? If so it would require some testing with different GPUs etc unless someone already has this info. (Alternatively we could keep this for now and later update the values later based on benchmarking)

Reply:
Yes, please include an explanation that for some years the loan data files have so many records that the memory consumption of the join, groupby, etc leads to an OOM situation on 16gb cards.